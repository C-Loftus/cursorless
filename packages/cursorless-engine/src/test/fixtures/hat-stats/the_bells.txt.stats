nTokens: 782

tokenSpark:
	▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▃▁▃▁▃▁▁▃▁▁▁▃▃▃▃▃▃▁▁▃▃▃▃▃▃▁▃▃▁▃▃▃▃▃▃▃▃▃▃▃▃
	▃▃▃▃▃▁▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▅▃▃▃▃▁▃▃▃▃▃▃▃▃▃▃▃▃▃▃
	▅▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▁▃▃▅▃▃▃▃▅▃▃▁▃▃▃▃▃▃▃▃▃▅▃▃▃▃▃▃▃▃▅
	▃▃▃▃▃▅▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▅▃▃▃▃▃▃▃▃▃▃▅▃▅▃▅▃▅▃▅▃▅▃▅▃▅▃▅▃
	▅▃▃▃▅▃▃▅▃▃▅▃▃▃▃▃▅▃▃▃▃▁▃▃▃▅▃▃▅▃▅▃▃▃▃▃▅▃▃▃▃▃▃▃▃▃▃▅▅▃▃▅▃▅▃▃▃▃▅▃
	▅▃▅▃▅▃▅▃▃▅▅▃▅▅▃▅▃▅▃▁▅▅▃▃▃▃▅▃▃▅▃▅▃▅▅▅▃▃▅▃▅▃▃▅▃▅▅▅▅▃▅▃▅▅▅▅▃▃▃▃
	▃▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▃▅▅▅▃▅▅▃▅▅▅▅▅▅▅▅▃▅▅▃▅▅▃▅▅▅▅▅▅▃▃▅▅▅▅▅▅▅▃█▅▅▅
	▅▅▅▃▅▅▅▃▅█▅▅▃█▅▅▅▃▅▅▅▃▅▅█▅▅▃▅▅▅▅▅▅▅▅▅▃▅▅▅▃▅▅▅█▅█▅█▅█▅█▅█▅▃▅▅
	▃▅▅▃▅▅▅▅▃▃▅▅▅▅▅▅▃▅▅▅▅▅▅▅▅▃▅▅▃▅▅▅▃▅▅▅█▅▅▃▅▅▅▅▃▅▅▅▅▅▅▃▅▅▅▅▅▅▅▅
	▅▅▅▅▃▅▅▃▃▅█▅▃▃▅▅▅▃▅▅▅█▅▅█▅▅▅█▅█▅█▅▅▅▅█▅▅▅▅▅▅▅▅▅▅▅▅▃▅▅▅▅▅▅▅▅▅
	▅▅▅▅▅▅▅▅▁▅▅▃▅▅▅▅▃▅▅▅█▅█▅█▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅█▅▅▅▃▃▅█▅█▅█▅
	▅▅▅▅▅█▅▅▅▅▅▅▅▅▅▅▃▃▅█▅█▅█▅▅▅▅▅▅█▅▅▅▅▅▅▅▅▅▅█▅█▅▅▅▅▅▅▅▅▃▃▅█▅█▅█
	▅▅▃█▃█▃█▅▅▅▅▅█▅▅▅▅▅▅▅▅▅▅█▅█▅▅▅▅▅▅▅▅█▅▅▅█▅█▅█▅▅▅█▅█▅▅▅▅▅▅▅▅▅▅
	▅▃

nHats:
	mean: 94%
	std: 0%
	min: 94%
	max: 94%
	spark: 0              25               50               75         █     100

nPenalty0:
	mean: 5%
	std: 0%
	min: 5%
	max: 5%
	spark: 0   █          25               50               75               100

nPenalty1:
	mean: 40%
	std: 0%
	min: 40%
	max: 40%
	spark: 0              25        █▄     50               75               100

nPenalty2:
	mean: 49%
	std: 0%
	min: 49%
	max: 49%
	spark: 0              25             █ 50               75               100
