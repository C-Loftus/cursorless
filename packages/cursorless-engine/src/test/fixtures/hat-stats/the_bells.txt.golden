_   _
The Bells
[-] [---]

 _  _        _ _   _    _   _  _ _       _   _
by Edgar Allan Poe 1809 – 1849 (public domain)
[] [---] [---] [-] [--] | [--] |[----] [----]|

__
I.
||

        _      _ _       _      A   _  A
        Hear the sledges with the bells—
        [--] [-] [-----] [--] [-] [---]|

                    _   A    _
                 Silver bells!
                 [----] [---]|

  _  A  _     _   _        A         A         AA
What a world of merriment their melody foretells!
[--] | [---] [] [-------] [---] [----] [-------]|

          A    B    _  _ A     A B     B
        How they tinkle, tinkle, tinkle,
        [-] [--] [----]| [----]| [----]|

           A   B   _  B    A   _  B
           In the icy air of night!
           [] [-] [-] [-] [] [---]|

        B     C   B      C        A
        While the stars that oversprinkle
        [---] [-] [---] [--] [----------]

        C   D      A   C    A  A  C
        All the heavens, seem to twinkle
        [-] [-] [-----]| [--] [] [-----]

              D D A           A      _
           With a crystalline delight;
           [--] | [---------] [-----]|

            B      B D E   E F   F
         Keeping time, time, time,
         [-----] [--]| [--]| [--]|

          A E  B    B  A      C  1
         In a sort of Runic rhyme,
         [] | [--] [] [---] [---]|

1   E         B         F   C      B     D
To the tintinabulation that so musically wells
[] [-] [-------------] [--] [] [-------] [---]

       C    2   C    2 D    3 E    4 F    5
       From the bells, bells, bells, bells,
       [--] [-] [---]| [---]| [---]| [---]|

                           A
               1    6 2    1 3    B
               Bells, bells, bells—
               [---]| [---]| [---]|

  D    3   _          B 4      A     C   1  4    A
  From the jingling and the tinkling of the bells.
  [--] [-] [------] [-] [-] [------] [] [-] [---]|

B B
II.
[]|

                                     A
           A 5        E   C     5    2
        Hear the mellow wedding bells,
        [--] [-] [----] [-----] [---]|

                  D     6    C
                 Golden bells!
                 [----] [---]|

  F  1   B    E   C       6           D       A  D
What a world of happiness their harmony foretells!
[--] | [---] [] [-------] [---] [-----] [-------]|

            B     B  2     C   F B
        Through the balmy air of night
        [-----] [-] [---] [-] [] [---]

          F   C  C     C   2    D      E
        How they ring out their delight!
        [-] [--] [--] [-] [---] [-----]|

                                       A
              C  3       C_A          D3
           From the molten-golden notes,
           [--] [-] [----]|[----] [---]|

                              A
               3    B  D   D  4
               And all in tune,
               [-] [-] [] [--]|

           1    4   _        E 1
           What a liquid ditty floats
           [--] | [----] [---] [----]

                                   A
     E   D   D   A  B    5       D 5 2     E   B
    To the turtle-dove that listens, while she gloats
    [] [-] [----]|[--] [--] [-----]| [---] [-] [----]

                E  4  D   F
               On the moon!
               [] [-] [--]|

           A                            B
          56 2     E    E     E    C    1
         Oh, from out the sounding cells,
         []| [--] [-] [-] [------] [---]|

3    6 C     3   D     C            4    1
What a gush of euphony voluminously wells!
[--] | [--] [] [-----] [----------] [---]|

               6   E  F     2
               How it swells!
               [-] [] [----]|

                F  F  F
               How it dwells
               [-] [] [----]

            F   F     E 3   5 1    C
           On the Future! how it tells
           [] [-] [----]| [-] [] [---]

                           A
            4   1   E      1    E
           Of the rapture that impels
           [] [-] [-----] [--] [----]

          1   2     D      1   3 F
         To the swinging and the ringing
         [] [-] [------] [-] [-] [-----]

                       B      B      B
            5   4   D  2  5   3  6   4
           Of the bells, bells, bells,
           [] [-] [---]| [---]| [---]|

            A        B      B      C      C
         2  1     E  5   F  6   1  1   2  2
         Of the bells, bells, bells, bells,
         [] [-] [---]| [---]| [---]| [---]|

                    C      C
                 3  3   4  4   5  C
               Bells, bells, bells—
               [---]| [---]| [---]|

      A              A              A
   3  2    F       2 2   D        6 3     6  4
  To the rhyming and the chiming of the bells!
  [] [-] [-----] [-] [-] [-----] [] [-] [---]|

2  C
III.
[-]|

              A
            1 4     F       F     1D
         Hear the loud alarum bells—
         [--] [-] [--] [----] [---]|

                    _       25
                 Brazen bells!
                 [----] [---]|

       A           C    C
6      1  4    2   5 1  6    3           1     36
What tale of terror, now, their turbulency tells!
[--] [--] [] [----]| [-]| [---] [--------] [---]|

           A
       4   3         3   3 5    E
       In the startled ear of night
       [] [-] [------] [-] [] [---]

                                         A
        6     2      1  1     5       F  1
       How they scream out their affright!
       [-] [--] [----] [-] [---] [------]|

          A                 A       D
          1  2            4 5      B1
         Too much horrified to speak,
         [-] [--] [-------] [] [---]|

                             D       D
            3   2    4   4   2 4     3
         They can only shriek, shriek,
         [--] [-] [--] [----]| [----]|

                       A     D
                   2   1   3 4
                  Out of tune,
                  [-] [] [--]|

   A                     A   A         A B       D
6  1    3              1 6   4      5  2 1     5 5
In a clamorous appealing to the mercy of the fire,
[] | [-------] [-------] [] [-] [---] [] [-] [--]|

   A                   A      A                      D
 4 2   5  _            1      1 6     5        E   6 6
In a mad expostulation with the deaf and frantic fire,
[] | [-] [-----------] [--] [-] [--] [-] [-----] [--]|

                          E       E       E
                  2   3   1   4   2   5   3
            Leaping higher, higher, higher,
            [-----] [----]| [----]| [----]|

            A    A                 E
            2    3   5         6   4
            With a desperate desire,
            [--] | [-------] [----]|

             A
          6  4      3        D
         And a resolute endeavor
         [-] | [------] [------]

           A A   B   A   A      E
           3E1   2   1   1   E  5
         Now—now to sit or never,
         [-]|[-] [] [-] [] [---]|

           A    A   A B
        6  5    1   3 3   F   B  F   4   D
       By the side of the pale-faced moon.
       [] [-] [--] [] [-] [--]|[---] [--]|

             AE B   A    F A    F A    A
             66 4   1    1 2    2 3    2
            Oh, the bells, bells, bells!
            []| [-] [---]| [---]| [---]|

            A    A   A      A     A      A
            4    5   2      2     2      1
            What a tale their terror tells
            [--] | [--] [---] [----] [---]

                   A        A
                   4    1   3
                  Of Despair!
                  [] [-----]|

         A    A      F   A      F  A  A   A
         5    1     63   2 1    4  2  3   4
       How they clang, and clash, and roar!
       [-] [--] [---]| [-] [---]| [-] [--]|

       A    A  A       A
       6    6  3       2   4
       What a horror they outpour
       [--] | [----] [--] [-----]

 A  B         A B                 AA
 3  1      5  5 5   2             45
On the bosom of the palpitating air!
[] [-] [---] [] [-] [---------] [-]|

       A    B   B  A             F
       2    2   1  2   5    C    5
       Yet the ear it fully knows,
       [-] [-] [-] [] [---] [---]|

            A   B      A   F
            4   3      1   6
            By the twanging,
            [] [-] [------]|

              A  B
              3  4  2
            And the clanging,
            [-] [-] [------]

          A  B        A    A B   A
          4  6        5    2 2   6    A
         How the danger ebbs and flows;
         [-] [-] [----] [--] [-] [---]|

       A    B    A             A
       3    5    6       3     3
       Yet the ear distinctly tells,
       [-] [-] [-] [--------] [---]

            A   B
            3   6  A
            In the jangling,
            [] [-] [------]

              A C   B
              4 1   1        E
            And the wrangling.
            [-] [-] [-------]|

        A  C   A            B      A
        5  2   5         D  3      3
       How the danger sinks and swells,
       [-] [-] [----] [---] [-] [----]

A  C           A  C    B       A  C   B      B C   A
5  3      E    6  4    2       4  5   4      1 6   6    F
By the sinking or the swelling in the anger of the bells—
[] [-] [-----] [] [-] [------] [] [-] [---] [] [-] [---]|

             B  D   B
             1  1   1    1
             Of the bells—
             [] [-] [---]|

     B  D   B      B      B      B
     2  2   2      3      4      5
     Of the bells, bells, bells, bells,
     [] [-] [---]  [---]  [---]  [---]

            B      C      C
            6      1      2    2
            Bells, bells, bells—
            [---]  [---]  [---]|

 A  D            A D            B D   C    A
 5  3      6     6 4   4        2 5   3    6
 In the clamor and the clangor of the bells!
 [] [-] [----] [-] [-] [-----] [] [-] [---]|

 FF
IV.
[]|

            B  D        A   B  C  C
            5  6        4   3  1  4    3
          Hear the tolling of the bells—
          [--] [-] [-----] [] [-] [---]|

                  B   C    B
                  1   5    1
                 Iron bells!
                 [--] [---]|

B    B     B  B     A             A       A        B
3    6     1  4     1     6       6       4    3   2
What a world of solemn thought their monody compels!
[--] | [---] [] [----] [-----] [---] [----] [-----]|

        B  E            B E     A
        1  1        5   5 2     2
        In the silence of the night,
        [] [-] [-----] [] [-] [---]

          B  A          E  C
          4  4    1     3  1
        How we shiver with affright
        [-] [] [----] [--] [------]

  C    A            A       B    B    A B
  2    5      6     2       6    2    5 3
  At the melancholy menace of their tone!
  [] [-] [--------] [----] [] [---] [--]|

        C           A     C   A
        1    2      1     3   4
        For every sound that floats
        [-] [---] [---] [--] [----]

           A   A  A   B        B       C
           3   6  2   5        1       4
        From the rust within their throats
        [--] [-] [--] [----] [---] [-----]

                 B  C A
                 3  5 3    1
                 Is a groan.
                 [] | [---]|

          B   B        C   E
          2   2 4     46   4   5     5
        And the people—ah, the people—
        [-] [-] [----]|[]  [-] [----]|

          A   D   B       B  E   A
          5   1   6     6 4  5   3
       They that dwell up in the steeple,
       [--] [--] [---] [] [] [-] [-----]

                 D     B
                 2     3
                 All alone,
                 [-] [---]

          B C         A  E        F
          3 1         4  6        1
        And who tolling, tolling, tolling,
        [-] [-] [-----]  [-----]  [-----]

          B    D  A        B
          5    3  4        4
          In that muffled monotone,
          [] [--] [-----] [------]

         C    D    B  B  A  B
         2    4    2  6  4  3
         Feel a glory in so rolling
         [--] | [---] [] [] [-----]

          B   C   A       B  D A
          5   2   3       4  5 5    6
          On the human heart a stone—
          [] [-] [---] [---] | [---]|

        A  B    C     A    B  C    A
        6  5    1     5    6  2    1
     They are neither man nor woman—
     [--] [-] [-----] [-] [-] [---]|

        B  B    C     C      C    A  A
        1  6    2     6      1    6  2
     They are neither brute nor human—
     [--] [-] [-----] [---] [-] [---]|

                 B  C  A
                 2  1  5     _
              They are Ghouls:
              [--] [-] [----]|

          B    C        F  A C     A
          4    3  F     2  6 3     5  B
        And their king it is who tolls;
        [-] [---] [--] [] [] [-] [---]|

        D    B C      C      C
        6    3 2      3      4
        And he rolls, rolls, rolls,
        [-] [] [---]  [---]  [---]

                    C
                    5
                    Rolls
                    [---]

             E A    C    F   D    B
             1 1    3    3   1    4
             A pæan from the bells!
             | [--] [--] [-] [---]|

            B  C      B  C     C
            5  4      3  2     4
          And his merry bosom swells
          [-] [-] [---] [---] [----]

               F    B A     C F   D    B
               4    4 2     4 5   2    5
             With the pæan of the bells!
             [--] [-] [--] [] [-] [---]|

            B C     A    E   C  B
            6 3     1    2   4  4    C
          And he dances, and he yells;
          [-] [] [----]  [-] [] [---]|

                    B   F      C
          1         1   6      5
          Keeping time, time, time,
          [-----] [--]  [--]  [--]

           A E   C   C  A     C
           6 3   6   5  4     5
          In a sort of Runic rhyme,
          [] | [--] [] [---] [---]

              C  C   A    C  D  D    A
              3  6   3    6  1  3    4
             To the pæan of the bells—
             [] [-] [--] [] [-] [---]|

               C   D  D
               4   2  4    A
               Of the bells:
               [] [-] [---]|

                    B    C     D
          2         2    6     1
          Keeping time, time, time,
          [-----] [--]  [--]  [--]

           B E   D   D  A      B
           1 4   1   1  5      5
          In a sort of Runic rhyme,
          [] | [--] [] [---] [---]

             C  D      D      D   B   A  A
             5  3      5      2   5   6  5
            To the throbbing of the bells—
            [] [-] [-------] [] [-] [---]|

          C   D  D      E      E    A
          6   4  6      1      2    6
          Of the bells, bells, bells—
          [] [-] [---]  [---]  [---]|

             D  D      D    D  D    B
             1  5      2    3  6    1  D
            To the sobbing of the bells;
            [] [-] [-----] [] [-] [---]|

                    B    D     D
          3         3    3     4
          Keeping time, time, time,
          [-----] [--]  [--]  [--]

            E  E
            5  1  4       5       6
            As he knells, knells, knells,
            [] [] [----]  [----]  [----]

          D  E     B  A       B
          5  6     6  6       4
          In a happy Runic rhyme,
          [] | [---] [---] [---]

             D  E        A  D  E  E    B
             2  2        6  4  3  3    1
            To the rolling of the bells—
            [] [-] [-----] [] [-] [---]|

          D   E  E      E      E    B
          3   4  4      5      6    2
          Of the bells, bells, bells—
          [] [-] [---]  [---]  [---]|

             D  E      D    D  E  F
             4  5      6    5  6  1
            To the tolling of the bells,
            [] [-] [-----] [] [-] [---]

      D   F  F      F      F      F    B
      5   1  2      3      4      5    3
      Of the bells, bells, bells, bells—
      [] [-] [---]  [---]  [---]  [---]|

              F       B      C   B
              6       6      1   4
              Bells, bells, bells—
              [---]  [---]  [---]|

   D  F    F       C  F   D        D  F    B
   6  2    1       1  3   2        6  4    2  2
  To the moaning and the groaning of the bells.
  [] [-] [-----] [-] [-] [------] [] [-] [---]|
